File: client-app/src/App/Models/Activity.ts
import { Profile } from "./profile";

export interface Activity {
  id: string;
  title: string;
  date: Date | null;
  description: string;
  category: string;
  city: string;
  venue: string;
  isCancelled: boolean,
  hostUserName: string,
  isHosting: boolean,
  isJoined: boolean,
  hostProfile?: Profile,
  attendees: Profile[]
}

export class Activity implements Activity {
  constructor(init?: ActivityFormValues) {
    Object.assign(this, init);
  }
}

export class ActivityFormValues {
  id?: string = undefined;
  title: string = '';
  date: Date | null = null;
  description: string = '';
  category: string = '';
  city: string = '';
  venue: string = '';

  constructor(values?: ActivityFormValues) {
    if (values) {
      this.id = values.id;
      this.title = values.title;
      this.date = values.date;
      this.description = values.description;
      this.category = values.category;
      this.city = values.city;
      this.venue = values.venue;
    }
  }
}

File: Domain/Activity.cs
namespace Domain;

public class Activity
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public DateTime Date { get; set; }
    public string Description { get; set; }
    public string Category { get; set; }
    public string City { get; set; }
    public string Venue { get; set; }
    public bool IsCancelled { get; set; }
    public ICollection<UserActivities> Attendees { get; set; } = new List<UserActivities>();
    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
}
